zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/UserControllerTest.java

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.rest.controller;
+
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.camunda.identity.usermanagement.CamundaUser;
+import io.camunda.identity.usermanagement.CamundaUserWithPassword;
+import io.camunda.identity.usermanagement.service.UserService;
+import io.camunda.zeebe.gateway.protocol.rest.CamundaUserDto;
+import io.camunda.zeebe.gateway.protocol.rest.CamundaUserWithPasswordDto;
+import io.camunda.zeebe.gateway.protocol.rest.SearchRequestDto;
+import io.camunda.zeebe.gateway.protocol.rest.UserSearchResponseDto;
+import io.camunda.zeebe.gateway.rest.GlobalControllerExceptionHandler;
+import io.camunda.zeebe.gateway.rest.controller.UserControllerTest.TestUserControllerApplication;
+import io.camunda.zeebe.gateway.rest.controller.usermanagement.UserController;
+import java.net.URI;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
+import org.springframework.boot.autoconfigure.orm.jpa.HibernateJpaAutoConfiguration;
+import org.springframework.boot.autoconfigure.security.servlet.SecurityAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ProblemDetail;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.test.web.reactive.server.WebTestClient;
+
+@SpringBootTest(
+    classes = {
+      TestUserControllerApplication.class,
+      UserController.class,
+      GlobalControllerExceptionHandler.class
+    },
+    webEnvironment = WebEnvironment.RANDOM_PORT)
+public class UserControllerTest {
```

Comment:
🔧 We should extend the user controller tests by cases for input validation, e.g. creating or updating a user without appropriate data. Those tests will verify and document the validation we have for the controller. Otherwise, it's cumbersome to identify issues later on regarding user requests that don't work as expected. With test cases documenting what is allowed and what not, we can easily see if the reported behavior is expected or not.

Reply:
As the input validation is currently done on the service level and tested there, and the controller tests cover that the controller methods return the desired response in case of validation exceptions, the current setup should be covered by tests. Wdyt?

Reply:
Yep, I think this suffices. I missed that the validation is done and tested in the service 👍